.gitlab/basic.gitlab-ci.yml

```diff
@@ -0,0 +1,172 @@
+      echo $jobid > "$WORKDIR/jobid_${jobid}"
+
+      partition=$(scontrol show jobid -dd $jobid | grep Partition)
+      export partition=$(echo $partition | cut -f2 -d'=' | cut -f1 -d' ')
+      
+      echo "Job $jobid submitted to partition $partition"
+      popd
+
+      echo "$partition" >> ./partition
+     
+      pushd $WORKDIR
+      res=1
+      while :;
+      do
+        if [[ "$(awk 'BEGIN{i=0}/BUILD_STATUS/{i++}END{print i}' output)" != "0" ]]; then
+          kill $tailpid
+          echo 'Last tail of build output:'
+          tail -n 200 output
+          res=$(grep BUILD_STATUS output | tail -n 1 | cut -f2 -d':')
+          break
+        fi
+        sleep 30
+      done
+      rm "$WORKDIR/jobid_${jobid}"
+      echo "Finished job with code $res"
+      exit $res
+
+.report-status:
+  image: mrnonz/alpine-git-curl:alpine3.16
+  variables:
+    GIT_STRATEGY: none
+    STATUS_PROJECT: ORNL/hykkt
+    STATUS_NAME: NotSet
+  extends:
+  - .pnnl_nonhpc_tags
+  script:
+    - |
+      set -x
+      if [[ ! -e partition ]]; then
+        echo "No partition file found"
+        export part="none"
+      else
+        export part=$(cat partition)
+      fi
+      
+      export newell_status="ppc64le/gcc@8.5.0/cuda@11.4/v100@70"
+      export deception_status="x86_64/gcc@9.1/cuda@11.4"
+     
+      if [[ "$part" == *"newell"*  ]]; then
+        export STATUS_NAME=$newell_status
+
+      elif [[ "$part" == *"a100"*  ]]; then
+        export gpu_arch=a100@80
+        export STATUS_NAME="$deception_status/$gpu_arch"
+
+      elif [[ "$part" == *"dl"*  ]]; then
+        gpu_arch=p100@60
+        export STATUS_NAME="$deception_status/$gpu_arch"
+
+      elif [[ "$part" == *"dlv"*  ]]; then
+        gpu_arch=v100@70
+        export STATUS_NAME="$deception_status/$gpu_arch"
+        
+      elif [[ "$part" == *"dlt"*  ]]; then
+        gpu_arch=RTX2080@75
+        export STATUS_NAME="$deception_status/$gpu_arch"
+        
+      else
+        echo "Unknown partition"
+        export STATUS_NAME="Unknown Partition"
+      fi
+      echo "GPU variable == $gpu_arch"
+      
+      curl -L \
+      -X POST \
+      -H @${GITHUB_CURL_HEADER}\
+       https://api.github.com/repos/${STATUS_PROJECT}/statuses/${CI_COMMIT_SHA} \
+       -d "{\"state\":\"${CI_JOB_NAME}\",\"target_url\":\"${CI_PIPELINE_URL}\",\"description\":\"${STATUS_NAME}\",\"context\":\"${MY_CLUSTER}\"}"
+  environment:
+    name: reporting-gitlab
+
+.report-pending:
+  image: mrnonz/alpine-git-curl:alpine3.16
+  variables:
+    GIT_STRATEGY: none
+    STATUS_PROJECT: ORNL/hykkt
+  extends: .pnnl_nonhpc_tags
+  script:
+   -  |
+      set -x
+      curl -L \
+      -X POST \
+      -H @${GITHUB_CURL_HEADER}\
+       https://api.github.com/repos/${STATUS_PROJECT}/statuses/${CI_COMMIT_SHA} \
+       -d "{\"state\":\"${CI_JOB_NAME}\",\"target_url\":\"${CI_PIPELINE_URL}\",\"context\":\"${MACHINE}\"}"
+  environment:
+    name: reporting-gitlab
```

Comment:
```suggestion
.status-base:
  image: mrnonz/alpine-git-curl:alpine3.16
  variables:
    GIT_STRATEGY: none
    STATUS_PROJECT: ORNL/hykkt
    STATUS_NAME: NotSet
  extends: .pnnl_nonhpc_tags
  environment:
    name: reporting-gitlab
    
.report-status:
  extends: .status-base
  stage: .post
  script:
    - |
      set -x
      if [[ ! -e partition ]]; then
        echo "No partition file found"
        export part="none"
      else
        export part=$(cat partition)
      fi
      
      export newell_status="ppc64le/gcc@8.5.0/cuda@11.4/v100@70"
      export deception_status="x86_64/gcc@9.1/cuda@11.4"
     
      if [[ "$part" == *"newell"*  ]]; then
        export STATUS_NAME=$newell_status

      elif [[ "$part" == *"a100"*  ]]; then
        export gpu_arch=a100@80
        export STATUS_NAME="$deception_status/$gpu_arch"

      elif [[ "$part" == *"dl"*  ]]; then
        gpu_arch=p100@60
        export STATUS_NAME="$deception_status/$gpu_arch"

      elif [[ "$part" == *"dlv"*  ]]; then
        gpu_arch=v100@70
        export STATUS_NAME="$deception_status/$gpu_arch"
        
      elif [[ "$part" == *"dlt"*  ]]; then
        gpu_arch=RTX2080@75
        export STATUS_NAME="$deception_status/$gpu_arch"
        
      else
        echo "Unknown partition"
        export STATUS_NAME="Unknown Partition"
      fi
      echo "GPU variable == $gpu_arch"
      
      curl -L \
      -X POST \
      -H @${GITHUB_CURL_HEADER}\
       https://api.github.com/repos/${STATUS_PROJECT}/statuses/${CI_COMMIT_SHA} \
       -d "{\"state\":\"${CI_JOB_NAME}\",\"target_url\":\"${CI_PIPELINE_URL}\",\"description\":\"${STATUS_NAME}\",\"context\":\"${MY_CLUSTER}\"}"

.report-pending:
  extends: .status-base
  stage: .pre
  script:
   -  |
      set -x
      curl -L \
      -X POST \
      -H @${GITHUB_CURL_HEADER}\
       https://api.github.com/repos/${STATUS_PROJECT}/statuses/${CI_COMMIT_SHA} \
       -d "{\"state\":\"${CI_JOB_NAME}\",\"target_url\":\"${CI_PIPELINE_URL}\",\"context\":\"${MY_CLUSTER}\"}"
```

Reply:
Just suggesting a way to consolidate things. I really think these could be combined into the same job. You can just throw a conditional based on the `CI_JOB_NAME` at the start of the first script, and then run with the curl post without the description.

I am also fine with leaving it as is, as it is more readable being separate IMO.

Reply:
I did change from `MACHINE` to `MY_CLUSTER` though as the difference didn't make sense, and keeping them the same makes some refactoring below easier...

Reply:
I think I prefer them to be separated. I don't think consolidating is a positive in this situation.

Reply:
Agreed, but I still suggested some important changes here as well to allow consolidation of other things.